projects/aws/bottlerocket-bootstrap/pkg/kubeadm/waiters.go

```diff
@@ -44,3 +45,18 @@ func killCmdAfterJoinFilesGeneration(cmd *exec.Cmd) error {
 	}
 	return utils.KillCmdAfterFilesGeneration(cmd, checkFiles)
 }
+
+// checkEbsInit checks the execution of ebs-init goroutine,
+// the goroutine will be stopped if timeout is reached
+func checkEbsInit(ctrl *EbsInitControl) {
+
+	select {
+	case <-ctrl.Timeout:
+		ctrl.Cancel()
+		fmt.Printf("Killing ebs-init, timeout reached \n")
+		return
+	case <-ctrl.OkChan:
+		fmt.Printf("Finished ebs-init \n")
+		return
+	}
```

Comment:
How does this work? I tested this out locally and the timeout just seems to cancel the context and print the line below but I still see the process reading the fs. 

Is this intended? If so, there is no need to even have a context in this scenario